### PR TITLE
Adds windows collection of failed conns.

### PR DESCRIPTION
**Pull request by derekwbrown**
This PR adds failed connections on Windows to the NPM product.

Adds new driver which detects failed connections on established connections.

Automated tests included .

### CODE REVIEW

**Review by clarkb7 — APPROVED** [review]

File: pkg/network/tracer/tracer_test.go [hunk 1]
```diff
@@ -1253,3 +1254,60 @@ func (s *TracerSuite) TestTCPDirection() {
 	conn = incomingConns[0]
 	assert.Equal(t, conn.Direction, network.INCOMING, "connection direction must be incoming: %s", conn)
 }
+
+func (s *TracerSuite) TestTCPFailureConnectionRefused() {
+	t := s.T()
+
+	checkSkipFailureConnectionsTests(t)
```
Comment (akarpz) [inline pkg/network/tracer/tracer_test.go:1261]: this will skip the Refused test on older kernels on linux, which we only want to do for the timeout test

Comment (derekwbrown) [reply pkg/network/tracer/tracer_test.go:1261]: 🙇

File: pkg/network/tracer/tracer_linux_test.go [hunk 2]
```diff
@@ -2368,15 +2368,21 @@ var failedConnectionsBuildModes = map[ebpftest.BuildMode]struct{}{
...
 
-func (s *TracerSuite) TestTCPFailureConnectionTimeout() {
-	t := s.T()
+func checkSkipFailureConnectionsTests(t *testing.T) {
 	if _, ok := failedConnectionsBuildModes[ebpftest.GetBuildMode()]; !ok {
 		t.Skip("Skipping test on unsupported build mode: ", ebpftest.GetBuildMode())
 	}
+
+}
+func (s *TracerSuite) TestTCPFailureConnectionTimeout() {
```
Comment (hmahmood) [inline pkg/network/tracer/tracer_linux_test.go:2377]: Missing line break.

**Review by hmahmood — APPROVED** [review]

**Pull request merged**